app/ui/expiryOptions.js

```diff
@@ -31,12 +31,12 @@ module.exports = function(state, emit) {
       counts,
       num => state.translate('downloadCount', { num }),
       value => {
-        const max = state.user.maxDownloads;
-        state.archive.dlimit = Math.min(value, max);
-        if (value > max) {
+        const selected = parseInt(value);
+        state.archive.dlimit = selected;
+        emit('render');
+        if (selected > parseInt(state.user.maxDownloads || '0')) {
+          console.log('Chosen max download count is larger than the allowed limit for anonymous users', selected)
```

Comment:
I don't think this message is correct. This must have been an oversight from previous changes.

There currently is no concept of 'anonymous' users, and the anonymous limits have been removed. Mozilla's intention was to implement FxA authentication but we've stripped this (as per https://gitlab.com/timvisee/send/-/issues/3).

Previously, `const max = state.user.maxDownloads;` was set to the anonymous limit if the user was not authenticated. In that case it made sense to check whether the download count was over their anonymous limit, showing the error. As I've mentioned, this does not make sense anymore.

I think, changing the message to something like this would be better:

```javascript
console.log('Chosen max download count is larger than the allowed limit', selected)
```

Also, a few lines lower, I think this can now be removed:

```javascript
emit('signup-cta', 'count');
```

Reply:
Actually this whole changeset for this file and the selectbox js can be reverted, it turns out only the `validate:` and `coerce:` change is needed in the config file at the bottom to fix the bug.

Reply:
Good question. Applying the changes I've suggested would probably make the UI logic better in the end. So I'd prefer approach A. It shouldn't break anything.